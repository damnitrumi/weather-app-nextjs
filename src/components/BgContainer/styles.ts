import styled, { css } from "styled-components";
import { BgContainerProps } from ".";

export const Wrapper = styled.article<Pick<BgContainerProps, "bgSelection">>`
  ${({ bgSelection }) => css`
    display: flex;
    flex-direction: column;
    background-image: url(${bgSelection});
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    width: 100%;
    height: 100%;
    margin: 0 auto;

    @media(min-width: 640px){
      flex-direction: row;
    }

    @media(min-width: 768px){
      aspect-ratio: 16/9;
      height: auto;
      max-width: 1000px;
      border-radius: 10px;
      overflow: hidden;

    }
  `}
`;
