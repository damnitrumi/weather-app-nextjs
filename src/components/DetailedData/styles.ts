import styled, { css } from "styled-components";

export const Wrapper = styled.section`
  ${() => css`
    padding:20px;
    min-width: 200px;
    backdrop-filter: blur(10px) saturate(180%);
    -webkit-backdrop-filter: blur(10px) saturate(180%);

    @media (min-width: 640px){
      flex-grow: 2;
    }

    @media (min-width: 1024px){
      padding: 40px;
    }
  `}
`;
