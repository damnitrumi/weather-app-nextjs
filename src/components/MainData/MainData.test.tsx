import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { MainData, MainDataProps } from ".";

const props: MainDataProps = {
  temp: 1,
  city: "City Name",
  time: "10:20",
  weekDay: "Friday",
  monthDay: "24",
  month: "Feb",
  year: "23",
  weather: "Rain",
  icon: "any",
};

describe("<MainData />", () => {
  it("should render elements with correct data", () => {
    renderTheme(<MainData {...props} />);

    expect(screen.getByText(`${props.temp}Â°`)).toBeInTheDocument();
    expect(screen.getByText(`${props.city}`)).toBeInTheDocument();
    expect(screen.getByText(`${props.weather}`)).toBeInTheDocument();
    expect(
      screen.getByText(
        `${props.time} - ${props.weekDay}, ${props.monthDay} ${props.month} '${props.year}`,
      ),
    ).toBeInTheDocument();

    expect(screen.getByRole("img")).toHaveAttribute(
      "src",
      `http://openweathermap.org/img/w/${props.icon}.png`,
    );
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<MainData {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
