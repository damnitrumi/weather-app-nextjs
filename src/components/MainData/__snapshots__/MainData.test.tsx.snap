// Vitest Snapshot v1

exports[`<MainData /> > should match snapshot 1`] = `
<section
  class="text-white px-4 py-32 flex items-end grow md:px-16 lg:px-24 lg:py-56"
>
  <div
    class="flex gap-5 items-end"
  >
    <p
      class=" text-[5.0rem] font-bold leading-none min-[500px]:text-[8.4rem]"
    >
      1
      Â°
    </p>
    <div
      class=""
    >
      <p
        class=" text-3xl font-medium min-[500px]:text-6xl"
      >
        City Name
      </p>
      <p
        class="text-lg pl-1 min-[500px]:text-[1.6rem]"
      >
        10:20
         - 
        Friday
        , 
        24
         
        Feb
         '
        23
      </p>
    </div>
    <div>
      <img
        alt="Weather Icon"
        src="http://openweathermap.org/img/w/any.png"
      />
      <p
        class="text-lg min-[500px]:text-[1.6rem]"
      >
        Rain
      </p>
    </div>
  </div>
</section>
`;
